heidrun/nara_json.rb

```diff
@@ -0,0 +1,575 @@
+def nara_catalog_uri(id)
+  "http://catalog.archives.gov/id/#{id.node}"
+end
+
+# See object and preview mappings
+def make_obj_uri(obj)
+  URI::escape(obj.node['@url'])  # Can have space characters
+end
+def make_obj_dcformat(obj)
+  obj.node['@mime']
+end
+def make_preview_uri(obj)
+  URI::escape(obj.node['@url'])  # Can also have space characters
+end
+def make_preview_dcformat(obj)
+  obj.node['@mime']
+end
+
+# Return a string suitable for sourceResource.contributor or
+# sourceResource.publisher.
+#
+# @see #make_contributor
+# @see #make_publisher
+# @param contributor_array [Krikri::JsonParser::Value]
+# @return [String, RDF::Literal]
+def contributor_term_name(contributor_array)
+  node = contributor_array.node
+  contributors = node.fetch('organizationalContributor',
+                            node['personalContributor'])
+  contributors = [contributors] unless contributors.is_a? Array
+
+  yield(contributors)
+
+  return contributors.first['contributor']['termName'] \
+    unless contributors.empty?
+
+  # FIXME:  It's not clear how to indicate the concept of "nothing" to
```

Comment:
My sense is that @anarchivist's first option is right. You should simply be filtering out values that are empty before this point. Here and elsewhere, there are a lot of empty nodes being created, leading to a bunch of data like this:

```turtle
<http://purl.org/dc/terms/contributor> [ a <http://www.europeana.eu/schemas/edm/Agent>];
    <http://purl.org/dc/terms/isPartOf> [
      a <http://purl.org/dc/dcmitype/Collection>;
      <http://purl.org/dc/terms/title> "General Records of the Department of State"
    ];
    <http://purl.org/dc/terms/language> [ a <http://www.w3.org/2004/02/skos/core#Concept>];
    <http://purl.org/dc/terms/publisher> [ a <http://www.europeana.eu/schemas/edm/Agent>];
    <http://purl.org/dc/terms/spatial> [ a <http://www.europeana.eu/schemas/edm/Place>];
```

This says that there is a contributor, place, agent, etc..., each with no properties, that are affiliated with this object.

There's a way to do this, but I'm digging through these to try to understand what's going on.